thirdparty/CMakeLists.txt

```diff
@@ -35,14 +35,15 @@ if(NOT spdlog_POPULATED)
 endif()
 
 ###################################################################################################
-# - thrust ----------------------------------------------------------------------------------------
+# - thrust/cub ------------------------------------------------------------------------------------
+
+message("Fetching Thrust")
 
 FetchContent_Declare(
     thrust
     GIT_REPOSITORY https://github.com/thrust/thrust.git
-    GIT_TAG        1.9.10
-    GIT_SHALLOW    true
-    PATCH_COMMAND COMMAND patch -p1 < "${CMAKE_CURRENT_SOURCE_DIR}/thrust-ret-if-fail.patch"
+    # August 28, 2020
+    GIT_TAG        52a8bda46c5c2128414d1d47f546b486ff0be2f0
```

Comment:
I guess a general comment is that it's nicer not to rely on arbitrary commits but rather on named version, but I suppose this is because you want to rely on some of the latest fixes, so I think it does make sense (but it'd be nice to replace it with the next release when that happens).

Reply:
Agreed.

Reply:
@germasch completely agree the commit makes me nervous and also prevents GIT_SHALLOW.

Reply:
From https://cmake.org/cmake/help/latest/module/ExternalProject.html:
> In general, however, specifying a commit hash should be preferred for a number of reasons:
> * If the local clone already has the commit corresponding to the hash, no git fetch needs to be performed to check for changes each time CMake is re-run. This can result in a significant speed up if many external projects are being used.
> * Using a specific git hash ensures that the main project’s own history is fully traceable to a specific point in the external project’s evolution. If a branch or tag name is used instead, then checking out a specific commit of the main project doesn’t necessarily pin the whole build to a specific point in the life of the external project. The lack of such deterministic behavior makes the main project lose traceability and repeatability.

Reply:
Using a git commit hash should generally always be preferred over a tag as it guarantees the code doesn't change out from underneath us (unless we're okay with that).

As far as using a commit that isn't tied to a specific release, I think we're okay as Thrust is header only and the exposure points of Thrust from RMM are pretty minimal.